modules/tests/kubecf-test.sh

```diff
@@ -58,6 +58,118 @@ wait_for_tests_pod() {
+    - ports:
+      - port: 53
+        protocol: UDP
+      - port: 53
+        protocol: TCP
+    - to:
+      - namespaceSelector: {}
+EOF
+      # For debugging, show what kube thinks of it.
+      kubectl describe networkpolicies \
+        --namespace "${KUBECF_NAMESPACE}" \
+        cats-internetless
+    else
+      # shellcheck disable=SC2005
+      echo "$(blue "Configure cluster network: Full access")"
+      # disable isolation
+      kubectl delete networkpolicies \
+        --namespace "${KUBECF_NAMESPACE}" \
+        cats-internetless
+    fi
+}
+
+create_cats_internetless_secret() {
+  info "Creating cats secret to use internetless CATS test suite"
+  cats_secret_name="$(
+    kubectl get qjobs -n "${KUBECF_NAMESPACE}" \
+    "${KUBECF_DEPLOYMENT_NAME}"-acceptance-tests -o json | \
+    jq -r '.spec.template.spec.template.spec.volumes[] | select(.name=="ig-resolved").secret.secretName')"
+
+  cats_secret="$(kubectl get secrets --export -n "${KUBECF_NAMESPACE}" "${cats_secret_name}" -o yaml)"
+
+  cats_updated_properties="$(
+    kubectl get secret -n "${KUBECF_NAMESPACE}" "${cats_secret_name}" -o json  | \
+    jq -r '.data."properties.yaml"' | base64 -d | \
+    yq w - "instance_groups.(name==acceptance-tests).jobs.(name==acceptance-tests).properties.acceptance_tests.include" '=internetless' | \
+    yq w - "instance_groups.(name==acceptance-tests).jobs.(name==acceptance-tests).properties.acceptance_tests.credhub_mode" '"off"' | \
+    base64 -w 0 )"
+
+  cats_updated="$(echo "$cats_secret" | yq w - 'data[properties.yaml]' $cats_updated_properties)"
+  cats_updated="$(echo "$cats_updated" | yq w - 'metadata.name' 'cats-internetless')"
+  kubectl apply -f <(echo "${cats_updated}") -n "${KUBECF_NAMESPACE}"
+}
+
+mount_cats_internetless_secret() {
+  original_volumes=$(
+    kubectl get qjob "${KUBECF_DEPLOYMENT_NAME}"-acceptance-tests --namespace "${KUBECF_NAMESPACE}" -o json | \
+    jq -r '.spec.template.spec.template.spec.volumes')
+
+  updated_volumes=$(echo ${original_volumes} | \
+    jq -r 'map((select(.name=="ig-resolved") | .secret.secretName) |= "cats-internetless")')
+
+  patch='{ "spec": { "template": { "spec": { "template": { "spec": { "volumes": '${updated_volumes}' } } } } } }'
+
+  kubectl patch qjob "${KUBECF_DEPLOYMENT_NAME}"-acceptance-tests \
+    --namespace "${KUBECF_NAMESPACE}" --type merge --patch "${patch}"
+}
+
+# Enables internet access again and mounte the original secret in the qjob
+# to enable other cats again.
+cleanup_cats_internetless() {
+  rv=$?
+  isolate_network 0
+  revert_patch='{ "spec": { "template": { "spec": { "template": { "spec": { "volumes": '${original_volumes}' } } } } } }'
+  echo "$(blue "Mounting the original secret in acceptance tests qjob")"
+  kubectl patch qjob "${KUBECF_DEPLOYMENT_NAME}"-acceptance-tests --namespace "${KUBECF_NAMESPACE}" --type merge --patch "${revert_patch}"
+  kubectl delete secret -n ${KUBECF_NAMESPACE} cats-internetless
+  trap "exit \$rv" EXIT
```

Comment:
This is to preserve the original exit code and not return whatever the cleanup function returns. I remember we did that some other place too but I don't remember if we used the same "trick". This one seems to work.